compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -391,28 +391,45 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     custom_span_label = true;
                 }
                 if static_sources.len() == 1 {
-                    let ty_str =
-                        if let Some(CandidateSource::Impl(impl_did)) = static_sources.get(0) {
-                            // When the "method" is resolved through dereferencing, we really want the
-                            // original type that has the associated function for accurate suggestions.
-                            // (#61411)
-                            let ty = tcx.at(span).type_of(*impl_did);
-                            match (&ty.peel_refs().kind(), &actual.peel_refs().kind()) {
-                                (ty::Adt(def, _), ty::Adt(def_actual, _)) if def == def_actual => {
-                                    // Use `actual` as it will have more `substs` filled in.
-                                    self.ty_to_value_string(actual.peel_refs())
-                                }
-                                _ => self.ty_to_value_string(ty.peel_refs()),
+                    let (ty_str, placeholders) = if let Some(CandidateSource::Impl(impl_did)) =
+                        static_sources.get(0)
+                    {
+                        // When the "method" is resolved through dereferencing, we really want the
+                        // original type that has the associated function for accurate suggestions.
+                        // (#61411)
+                        let ty = tcx.at(span).type_of(*impl_did);
+                        match (&ty.peel_refs().kind(), &actual.peel_refs().kind()) {
+                            (ty::Adt(def, _), ty::Adt(def_actual, substs)) if def == def_actual => {
+                                // If there are any inferred arguments, (`{integer}`), we shouldn't mark
+                                // this as machine-applicable.
+                                let placeholders = substs
+                                    .iter()
+                                    .filter_map(|arg| {
+                                        if let GenericArgKind::Type(ty) = arg.unpack() {
+                                            Some(ty)
+                                        } else {
+                                            None
+                                        }
+                                    })
+                                    .any(|ty| matches!(ty.kind(), ty::Infer(_)));
```

Comment:
Consider making use of `is_suggestable` here instead of only checking for `ty::Infer`.